app/src/main/java/foi/air/szokpt/helpers/RegistrationHandler.kt

```diff
@@ -0,0 +1,31 @@
+package foi.air.szokpt.helpers
+
+import android.util.Log
+import hr.foi.air.core.register.RegistrationBody
+import hr.foi.air.core.network.ResponseListener
+import hr.foi.air.core.network.models.ErrorResponseBody
+import hr.foi.air.core.network.models.SuccessfulResponseBody
+import hr.foi.air.core.register.RegistrationOutcomeListener
+import hr.foi.air.szokpt.ws.models.RegistrationResponse
+import hr.foi.air.szokpt.ws.request_handlers.RegistrationRequestHandler
+
+class RegistrationHandler() {
+    fun register(jwtToken: String, registrationBody: RegistrationBody, registrationListener: RegistrationOutcomeListener) {
+        val registrationRequestHandler = RegistrationRequestHandler(jwtToken, registrationBody)
+        registrationRequestHandler.sendRequest(
+            object : ResponseListener<RegistrationResponse> {
+                override fun onSuccessfulResponse(response: SuccessfulResponseBody<RegistrationResponse>) {
+                    registrationListener.onSuccessfulRegistration(response.message)
+                }
+
+                override fun onErrorResponse(response: ErrorResponseBody) {
+                    registrationListener.onFailedRegistration(response.message  ?: "Registration error occurred")
+                }
+
+                override fun onNetworkFailure(t: Throwable) {
+                    registrationListener.onFailedRegistration("Could not connect to network.")
+                }
```

Comment:
Potencijalno se ovdje može dodati malo više konteksta korisniku o tomu što je specifičan problem ili što ga je izazvalo.